gym-unity/gym_unity/envs/__init__.py

```diff
@@ -71,27 +72,35 @@ def __init__(
         self.name = list(self._env.behavior_specs.keys())[0]
         self.group_spec = self._env.behavior_specs[self.name]
 
-        if use_visual and self._get_n_vis_obs() == 0:
+        if self._get_n_vis_obs() == 0 and self._get_vec_obs_size() == 0:
             raise UnityGymException(
-                "`use_visual` was set to True, however there are no"
-                " visual observations as part of this environment."
+                "There are no observations provided by the environment. "
             )
-        self.use_visual = self._get_n_vis_obs() >= 1 and use_visual
 
-        if not use_visual and uint8_visual:
+        if not self._get_n_vis_obs() >= 1 and uint8_visual:
             logger.warning(
                 "`uint8_visual was set to true, but visual observations are not in use. "
                 "This setting will not have any effect."
             )
         else:
             self.uint8_visual = uint8_visual
 
-        if self._get_n_vis_obs() > 1 and not self._allow_multiple_visual_obs:
+        if self._get_n_vis_obs() > 1 and not self._allow_multiple_obs:
```

Comment:
I think you can combine these two warnings and just check `self._get_n_vis_obs() + self._get_vec_obs_size() > 2`